app/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnector.scala

```diff
@@ -36,12 +35,12 @@ class TaxFreeAmountComparisonConnector @Inject()(val httpHandler: HttpHandler, s
 
   def taxFreeAmountComparisonUrl(nino: String): String = s"$serviceUrl/tai/$nino/tax-account/tax-free-amount-comparison"
 
-  def taxFreeAmountComparison(nino: Nino)(implicit hc: HeaderCarrier): Future[TaiResponse] =
-    httpHandler.getFromApiV2(taxFreeAmountComparisonUrl(nino.nino)) map (
-      json => TaiSuccessResponseWithPayload((json \ "data").as[TaxFreeAmountComparison])
-    ) recover {
+  def taxFreeAmountComparison(nino: Nino)(implicit hc: HeaderCarrier): Future[TaxFreeAmountComparison] =
+    httpHandler.getFromApiV2(taxFreeAmountComparisonUrl(nino.nino)).map { json =>
+      (json \ "data").as[TaxFreeAmountComparison]
+    } recover {
       case NonFatal(e) =>
-        logger.warn(s"Couldn't retrieve taxFreeAmountComparison for $nino with exception:${e.getMessage}")
-        TaiTaxAccountFailureResponse(e.getMessage)
+        logger.warn(s"Couldn't retrieve taxFreeAmountComparison for $nino with exception: ${e.getMessage}")
+        throw e
```

Comment:
I know this is what we were doing before but it seems bad. I presume there is a recover higher up that will capture an exception?

Reply:
if the intention is to have a warn. I don't see it a bad practise (as long as we wanna warn here)